bnd-maven-plugin-parent/bnd-maven-plugin/src/main/java/org/bndtools/maven/MavenVersion.java

```diff
@@ -0,0 +1,63 @@
+package org.bndtools.maven;
+
+/**
+ * Represents a Maven version with the ability to translate to a bnd version,
+ * with SNAPSHOT being replaced by bnd's ${tstamp} macro.
+ * 
+ * @author Neil Bartlett <neil.bartlett@paremus.com>
+ */
+public class MavenVersion {
+
+	private static final String SNAPSHOT = "SNAPSHOT";
+	private static final String TSTAMP = "${tstamp}";
+
+	private int major;
+	private int minor;
+	private int micro;
+	private String qualifier;
+	
+	public MavenVersion(String versionString) throws IllegalArgumentException {
```

Comment:
It's not so easy to copy because I can't preserve "${tstamp}" inside an actual OSGi Version object.

I like the rule that the version must be parseable as an OSGi version, and I think my code achieves that.